File: src/main/java/com/anod/appwatcher/installed/ImportItemAnimator.java
package com.anod.appwatcher.installed;

import android.animation.ArgbEvaluator;
import android.animation.ObjectAnimator;
import android.graphics.Color;
import android.support.annotation.ColorInt;
import android.support.annotation.NonNull;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;

import java.util.List;

/**
 * @author algavris
 * @date 24/04/2016.
 */
class ImportItemAnimator extends DefaultItemAnimator {

    @Override
    public boolean animateChange(RecyclerView.ViewHolder oldHolder, RecyclerView.ViewHolder newHolder, int fromX, int fromY, int toX, int toY) {
        oldHolder.itemView.clearAnimation();
        newHolder.itemView.clearAnimation();

        if (newHolder instanceof ImportAppViewHolder) {
            ImportAppViewHolder holder = (ImportAppViewHolder) newHolder;
            int status = holder.status();
            if (status == ImportDataProvider.STATUS_IMPORTING) {
                Animation anim = new AlphaAnimation(0.2f, 1.0f);
                anim.setDuration(500);
                anim.setRepeatMode(Animation.REVERSE);
                anim.setRepeatCount(Animation.INFINITE);
                holder.itemView.startAnimation(anim);
            } else if (status == ImportDataProvider.STATUS_DONE) {
                animateColor(holder.itemView, Color.TRANSPARENT, holder.themeAccent);
            } else if (status == ImportDataProvider.STATUS_ERROR) {
                animateColor(holder.itemView, Color.TRANSPARENT, holder.materialRed);
            }
        }

        return false;
    }

    private void animateColor(View view,@ColorInt int startColor, @ColorInt int endColor)
    {
        final ObjectAnimator animator = ObjectAnimator.ofObject(
                view, "backgroundColor" , new ArgbEvaluator(), startColor, endColor
        );
        animator.setDuration(300);
        animator.start();
    }

    @Override
    public boolean canReuseUpdatedViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, @NonNull List<Object> payloads) {
        return true;
    }
}

File: src/main/java/com/anod/appwatcher/installed/ImportItemAnimator.kt
package com.anod.appwatcher.installed

import android.animation.ArgbEvaluator
import android.animation.ObjectAnimator
import android.graphics.Color
import android.support.annotation.ColorInt
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.animation.AlphaAnimation
import android.view.animation.Animation

/**
 * @author algavris
 * *
 * @date 24/04/2016.
 */
internal class ImportItemAnimator : DefaultItemAnimator() {

    override fun animateChange(oldHolder: RecyclerView.ViewHolder, newHolder: RecyclerView.ViewHolder, fromX: Int, fromY: Int, toX: Int, toY: Int): Boolean {
        oldHolder.itemView.clearAnimation()
        newHolder.itemView.clearAnimation()

        if (newHolder is ImportAppViewHolder) {
            val status = newHolder.status()
            when (status) {
                ImportResourceProvider.STATUS_IMPORTING -> {
                    val anim = AlphaAnimation(0.2f, 1.0f)
                    anim.duration = 500
                    anim.repeatMode = Animation.REVERSE
                    anim.repeatCount = Animation.INFINITE
                    newHolder.itemView.startAnimation(anim)
                }
                ImportResourceProvider.STATUS_DONE -> animateColor(newHolder.itemView, Color.TRANSPARENT, newHolder.themeAccent)
                ImportResourceProvider.STATUS_ERROR -> animateColor(newHolder.itemView, Color.TRANSPARENT, newHolder.materialRed)
            }
        }

        return false
    }

    private fun animateColor(view: View, @ColorInt startColor: Int, @ColorInt endColor: Int) {
        val animator = ObjectAnimator.ofObject(
                view, "backgroundColor", ArgbEvaluator(), startColor, endColor
        )
        animator.duration = 300
        animator.start()
    }

    override fun canReuseUpdatedViewHolder(viewHolder: RecyclerView.ViewHolder, payloads: List<Any>): Boolean {
        return true
    }
}
